docs/rules/object-curly-newline.md

```diff
@@ -0,0 +1,478 @@
+/*eslint object-curly-newline: ["error", "never"]*/
+/*eslint-env es6*/
+
+let a = {
+};
+let b = {
+    foo: 1
+};
+let c = {
+    foo: 1, bar: 2
+};
+let d = {
+    foo: 1,
+    bar: 2
+};
+let e = {
+    foo: function() {
+        dosomething();
+    }
+};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {
+    g, h
+} = obj;
+let {
+    i,
+    j
+} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+Examples of **correct** code for this rule with the `"never"` option:
+
+```js
+/*eslint object-curly-newline: ["error", "never"]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {foo: 1, bar: 2};
+let d = {foo: 1,
+    bar: 2};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {} = obj;
+let {f} = obj;
+let {g, h} = obj;
+let {i,
+    j} = obj;
+let {k = function() {
+    dosomething();
+}} = obj;
+```
+
+### multiline
+
+Examples of **incorrect** code for this rule with the default `{"multiline": true}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"multiline": true}]*/
+/*eslint-env es6*/
+
+let a = {
+};
+let b = {
+    foo: 1
+};
+let c = {
+    foo: 1, bar: 2
+};
+let d = {foo: 1,
+    bar: 2};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {
+    g, h
+} = obj;
+let {i,
+    j} = obj;
+let {k = function() {
+    dosomething();
+}} = obj;
+```
+
+Examples of **correct** code for this rule with the default `{"multiline": true}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"multiline": true}]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {foo: 1, bar: 2};
+let d = {
+    foo: 1,
+    bar: 2
+};
+let e = {
+    foo: function() {
+        dosomething();
+    }
+};
+
+let {} = obj;
+let {f} = obj;
+let {g, h} = obj;
+let {
+    i,
+    j
+} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+### minProperties
+
+Examples of **incorrect** code for this rule with the `{"minProperties": 2}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"minProperties": 2}]*/
+/*eslint-env es6*/
+
+let a = {
+};
+let b = {
+    foo: 1
+};
+let c = {foo: 1, bar: 2};
+let d = {foo: 1,
+    bar: 2};
+let e = {
+    foo: function() {
+        dosomething();
+    }
+};
+
+let {
+} = obj;
+let {
+    f
+} = obj;
+let {g, h} = obj;
+let {i,
+    j} = obj;
+let {
+    k = function() {
+        dosomething();
+    }
+} = obj;
+```
+
+Examples of **correct** code for this rule with the `{"minProperties": 2}` option:
+
+```js
+/*eslint object-curly-newline: ["error", {"minProperties": 2}]*/
+/*eslint-env es6*/
+
+let a = {};
+let b = {foo: 1};
+let c = {
+    foo: 1, bar: 2
+};
+let d = {
+    foo: 1,
+    bar: 2
+};
+let e = {foo: function() {
+    dosomething();
+}};
+
+let {} = obj;
+let {f} = obj;
+let {
+    g, h
+} = obj;
+let {
+    i,
+    j
+} = obj;
+let {k = function() {
+    dosomething();
+}} = obj;
+```
+
+### a combination of multiline and minProperties
```

Comment:
http://eslint.org/docs/rules/key-spacing#singleline-and-multiline
http://eslint.org/docs/rules/no-unused-expressions#allowshortcircuit-and-allowternary